### PR TITLE
Migrate Upload store to redux

### PR DESCRIPTION
**Pull request by psinghal20**
Migrates Upload Store to redux.
Uses `sortByKey` method for sorting the uploads.
Screenshot: 
uploadstore

### CODE REVIEW

File: app/assets/javascripts/reducers/uploads.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  usage_count: true,
+  date: true,
+  uploader: true
+};
+
+export default function uploads(state = initialState, action) {
+  switch (action.type) {
+    case RECEIVE_UPLOADS: {
+      const dataUploads = action.data.course.uploads;
+      //Intial sorting by file name
```
Comment (ragesoss) [inline app/assets/javascripts/reducers/uploads.js]: Initial sorting should be by date.

File: app/assets/javascripts/reducers/uploads.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const initialState = {
+  uploads: [],
+  sortKey: null,
+};
+
+const SORT_DESCENDING = {
+  usage_count: true,
+  date: true,
+  uploader: true
```
Comment (ragesoss) [inline app/assets/javascripts/reducers/uploads.js]: Username should not be descending.

**Comment by ragesoss** [comment]
This looks mostly good. Sorting by clicking the column header is broken.

The sorting for usage count is buggy: it should put all the entries with `null` count at the end, but they end up at the beginning. Also, Date sorting should be added to sorting pulldown. (Both of those bugs are already present in the current version, but now's a good time to fix them.)

File: app/assets/javascripts/utils/model_utils.js [hunk 3]
```diff
@@ -28,10 +28,17 @@ export const sortByKey = (models, sortKey, previousKey = null, desc = false) =>
     newKey = sortKey;
   }
```
Comment (ragesoss) [inline app/assets/javascripts/utils/model_utils.js]: Documentation of this would be helpful.

**Comment by psinghal20** [comment]
@ragesoss I have added the documentation and the failing Travis build seems to be a random failure as the superset build is passing.

**Pull request merged**